### PR TITLE
feat: add xz compression

### PR DESCRIPTION
**Pull request by unickorn**
https://github.com/appwrite/appwrite/issues/4003

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -34,20 +35,41 @@ RUN git clone --recursive --depth 1 --branch $PHP_ZSTD_VERSION https://github.co
...
+    && rm xz.tar.xz \
+    && ( \
+        cd xz-${PHP_XZ_VERSION} \
+        && ./configure \
+        && make \
+        && make install \
+    ) \
+    && rm -r xz-${PHP_XZ_VERSION}
+
+RUN git clone https://github.com/codemasher/php-ext-xz.git \
```
Comment (stnguyen90) [inline Dockerfile:51]: Let's use release 1.1.2 instead of the default branch.

File: tests/Storage/Compression/Algorithms/XZTest.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Utopia\Tests\Compression\Algorithms;
```
Comment (stnguyen90) [inline tests/Storage/Compression/Algorithms/XZTest.php:3]: Would you please use:

```suggestion
namespace Tests\Storage\Compression\Algorithms;
```
to comply with psr4 autoload standards.

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
Local tests look good:

image

**Comment by stnguyen90** [comment]
@Unickorn Great PR! 🤯 We left some comments during the review, please check them out.

**Review by stnguyen90 — APPROVED** [review]
Tests are still passing!

image

@Unickorn Hey 👋 awesome work on your PR! We've approved your work and it'll be merged soon!

**Comment by christyjacob4** [comment]
Please sync your branch with `main` . It has fixes for the failing CI tests

**Comment by christyjacob4** [comment]
@unickorn please resolve the merge conflicts

File: Dockerfile [hunk 3]
```diff
@@ -35,6 +37,22 @@ RUN git clone --recursive --depth 1 --branch $PHP_ZSTD_VERSION https://github.co
...
+    && ( \
+        cd xz-${PHP_XZ_VERSION} \
+        && ./configure \
+        && make \
+        && make install \
+    ) \
+    && rm -r xz-${PHP_XZ_VERSION}
+
+RUN git clone https://github.com/codemasher/php-ext-xz.git --branch ${PHP_EXT_XZ_VERSION} \
+  && cd php-ext-xz \
```
Comment (christyjacob4) [inline Dockerfile:54]: This looks like a leftover. Can you please check

**Comment by christyjacob4** [comment]
THANK YOU! All changes merged 🥳

Please reach out to me on our [Discord server](https://appwrite.io/discord) if you would like to claim your Appwrite swags! As a way of saying thank you, we would also love to invite you to join the Appwrite organization on GitHub. Please share your GitHub username with us on Discord.  

You can accept the invite by visiting https://github.com/orgs/appwrite/invitation. By joining our team, you will officially be an Appwrite maintainer on GitHub.

You can change your membership [visibility settings](https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/publicizing-or-hiding-organization-membership), so your new Appwrite team membership badge will show up on your personal GitHub profile.

Please feel free to look for more PRs you might be interested in helping with on our long list of [Hacktoberfest friendly issues](https://github.com/search?q=org%3Aappwrite+org%3Autopia-php+org%3Aopen-runtimes+label%3Ahacktoberfest+created%3A%3E2021-01-01+state%3Aopen&type=issues&s=updated&o=asc) and help make Appwrite better :)

**Pull request merged**